KnockKnock-iOS/Scenes/Challenge/ChallengeViewController.swift

```diff
@@ -24,37 +24,71 @@ final class ChallengeViewController: BaseViewController<ChallengeView> {
       $0.delegate = self
       $0.dataSource = self
     }
+    self.containerView.sortChallengeButton.addTarget(self, action: #selector(tapSortChallengeButton(_:)), for: .touchUpInside)
+  }
+
+  func setNavigationItem() {
+    let searchBarButtonItem = UIBarButtonItem(barButtonSystemItem: .search, target: self, action: nil)
+
+    self.navigationItem.title = "챌린지"
+    self.navigationItem.rightBarButtonItem = searchBarButtonItem
+  }
+
+  @objc func tapSortChallengeButton(_ sender: UIButton) {
+    let bottomSheetViewController = BottomSheetViewController()
+    bottomSheetViewController.modalPresentationStyle = .overFullScreen
+    self.present(bottomSheetViewController, animated: false, completion: nil)
   }
 }
 
 extension ChallengeViewController: ChallengeViewProtocol {
   func fetchChallenges(challenges: [Challenge]) {
     self.challenges = challenges
-    self.containerView.tableView.reloadData()
+    self.containerView.challengeCollectionView.reloadData()
   }
 }
 
-extension ChallengeViewController: UITableViewDataSource {
-  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+extension ChallengeViewController: UICollectionViewDataSource {
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
     return self.challenges.count
   }
-  
-  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-    let cell = tableView.dequeueCell(withType: ChallengeCell.self, for: indexPath)
+
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! ChallengeCell
     let challenge = self.challenges[indexPath.row]
+    cell.backgroundColor = .white
     cell.bind(data: challenge)
     return cell
   }
 }
 
-extension ChallengeViewController: UITableViewDelegate {
-  
+extension ChallengeViewController: UICollectionViewDelegateFlowLayout {
+  func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+    return CGSize(width: (self.containerView.frame.width - 40), height: (self.containerView.frame.height - 120) / 2)
+  }
+  func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+    return 40
+  }
+  func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+    return 20
+  }
+}
+
+extension ChallengeViewController: UICollectionViewDelegate {
+  func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+    self.navigationController?.pushViewController(ChallengeDetailViewController(), animated: true)
```

Comment:
```suggestion
    let view = ChallengeDetailViewController()
    view.hidesBottomBarWhenPushed = true
    self.navigationController?.pushViewController(view, animated: true)
```

이런식으로 바텀 탭바를 제거 가능합니다.
그리고 작성해주신 방법과 같이 뷰를 이동하는것은 좋지 않아서 나중에는 제가 선언 해놓은 router를 통해서 뷰의 이동이 실행 되도록 수정 부탁드립니다.

Reply:
일단 제안해주신 방법으로 하고 다음 작업시 router를 통해 이동되도록 반영해두도록하겠습니다!